python/lsst/daf/persistence/posixStorage.py

```diff
@@ -348,5 +417,159 @@ def v1RepoExists(root):
+        path : string
+            The path to the file within the root directory.
+        searchParents : bool, optional
+            For Butler v1 repositories only, if true and a _parent symlink
+            exists, then the directory at _parent will be searched if the file
+            is not found in the root repository. Will continue searching the
+            parent of the parent until the file is found or no additional
+            parent exists.
+        """
+        # Separate path into a root-equivalent prefix (in dir) and the rest
+        # (left in path)
+        rootDir = root
+        # First remove trailing slashes (#2527)
+        while len(rootDir) > 1 and rootDir[-1] == '/':
+            rootDir = rootDir[:-1]
+
+        if path.startswith(rootDir + "/"):
+            # Common case; we have the same root prefix string
+            path = path[len(rootDir + '/'):]
+            pathPrefix = rootDir
+        elif rootDir == "/" and path.startswith("/"):
+            path = path[1:]
+            pathPrefix = None
+        else:
+            # Search for prefix that is the same as root
+            pathPrefix = os.path.dirname(path)
+            while pathPrefix != "" and pathPrefix != "/":
+                if os.path.realpath(pathPrefix) == os.path.realpath(root):
+                    break
+                pathPrefix = os.path.dirname(pathPrefix)
+            if pathPrefix == "/":
+                path = path[1:]
+            elif pathPrefix != "":
+                path = path[len(pathPrefix)+1:]
+
+        # Now search for the path in the root or its parents
+        # Strip off any cfitsio bracketed extension if present
+        strippedPath = path
+        pathStripped = None
+        firstBracket = path.find("[")
+        if firstBracket != -1:
+            strippedPath = path[:firstBracket]
+            pathStripped = path[firstBracket:]
+
+        dir = rootDir
+        while True:
+            paths = glob.glob(os.path.join(dir, strippedPath))
+            if len(paths) > 0:
+                if pathPrefix != rootDir:
+                    paths = [p[len(rootDir+'/'):] for p in paths]
+                if pathStripped is not None:
+                    paths = [p + pathStripped for p in paths]
+                return paths
+            if searchParents:
+                dir = os.path.join(dir, "_parent")
+                if not os.path.exists(dir):
+                    return None
+            else:
+                return None
+
+    @staticmethod
+    def prepOutputRootRepos(outputRoot, root):
+        # Path manipulations are subject to race condition
+        if not os.path.exists(outputRoot):
+            try:
+                os.makedirs(outputRoot)
+            except OSError as e:
+                if not e.errno == errno.EEXIST:
+                    raise
+            if not os.path.exists(outputRoot):
+                raise RuntimeError("Unable to create output repository '%s'" % (outputRoot,))
```

Comment:
This is not LSST-specific, Python docs says "This method of string formatting is the new standard in Python 3, and should be preferred to the % formatting":
https://docs.python.org/2/library/stdtypes.html#str.format

One issue is that python `logging` does not support new-style formatting (but `lsst.log` does since recently).